src/test/kotlin/com/gamenews/ApplicationTest.kt

```diff
@@ -52,4 +62,352 @@ class ApplicationTest {
+            )
+        }
+
+        // ASSERT
+        assertEquals(HttpStatusCode.BadRequest, response.status)
+        assertTrue(response.bodyAsText().contains("Sorry, this article no longer exists"))
+    }
+
+    @Test
+    fun `deleting an article succeeds and returns HTTPS Found status `() = testApplication {
+        // SET UP
+        application {
+            configureRouting(controller)
+            configureTemplating()
+        }
+        val title = "Deleted Article"
+        val body = "What did I do to deserve this?!"
+
+        val testArticle = Article.newEntry(title, body)
+
+        // Mock db to get an article but doesn't find it
+        coEvery { mockDB.getArticleById(testArticle.id) } returns testArticle
+        coEvery { mockDB.deleteArticle(testArticle.id) } returns true
+
+        // DO
+        val response = client.delete("/articles/${testArticle.id}") {
+            setBody(MultiPartFormDataContent(formData { }))
+        }
+
+        // ASSERT
+        assertEquals(HttpStatusCode.Found, response.status)
+    }
+
+    @Test
+    fun `deleting an article fails to delete and returns HTTPS Not Modified status `() = testApplication {
+        // SET UP
+        application {
+            configureRouting(controller)
+            configureTemplating()
+        }
+        val title = "I can't be deleted"
+        val body = "Evil laughs, muahahahahaha!"
+
+        val testArticle = Article.newEntry(title, body)
+
+        // Mock db to get an article but doesn't find it
+        coEvery { mockDB.getArticleById(testArticle.id) } returns testArticle
+        coEvery { mockDB.deleteArticle(testArticle.id) } returns false
+
+        // DO
+        val response = client.delete("/articles/${testArticle.id}") {
+            setBody(MultiPartFormDataContent(formData { }))
+        }
+
+        // ASSERT
+        assertEquals(HttpStatusCode.NotModified, response.status)
+        assertTrue(response.bodyAsText().contains("Deletion of the article was not successful!"))
+    }
+
+    @Test
+    fun `deleting an article fails due to article not found and returns HTTPS Bad Request status `() = testApplication {
+        // SET UP
+        application {
+            configureRouting(controller)
+            configureTemplating()
+        }
+        val title = "I don't exist"
+        val body = "I'm not real"
+
+        val testArticle = Article.newEntry(title, body)
+
+        // Mock db to get an article but doesn't find it
+        coEvery { mockDB.getArticleById(testArticle.id) } returns null
+
+        // DO
+        val response = client.delete("/articles/${testArticle.id}") {
+            setBody(MultiPartFormDataContent(formData { }))
+        }
+
+        // ASSERT
+        assertEquals(HttpStatusCode.BadRequest, response.status)
+        assertTrue(
+            response.bodyAsText().contains("Sorry, someone beat you to the punch. We couldn't find that article")
+        )
+    }
 }
+
+// SET UP
+
+// DO
+
+// ASSERT
```

Comment:
Delete

Reply:
I am surprised this was not found by detekt. I would have expected this to be caught by the tool.

Reply:
Done!